### PR TITLE
Add voronoi [09]

### PR DESCRIPTION
**Pull request by AbelVM**
- [x] All declared geometries are `geometry(Geometry, 4326)` for general geoms, or `geometry(Point, 4326)`
- [x] Existing functions in crankshaft python library called from the extension are kept at least from version N to version N+1 (to avoid breakage during upgrades).
- [x] Docs for public-facing functions are written
- [x] New functions follow the naming conventions: `CDB_NameOfFunction`. Where internal functions begin with an underscore `_`.
- [x] If appropriate, new functions accepts an arbitrary query as an input (see [Crankshaft Issue #6](https://github.com/CartoDB/crankshaft/issues/6) for more information)

### CODE REVIEW

**Comment by saleiva** [comment]
This is great. @AbelVM can you add a little description of the parameters needed for this analysis so I can create the UI?

**Comment by AbelVM** [comment]
https://github.com/CartoDB/crankshaft/blob/613cdb4c6fdd1dd0bad6d86fe4a80722075a7c12/doc/09_voronoi.md

**Comment by saleiva** [comment]
ok, will wait! :)

**Comment by AbelVM** [comment]
Fixed

**Comment by AbelVM** [comment]
selection_003

**Comment by AbelVM** [comment]
I've tested the code back and forth, and it works as is. The message in tests looks wrong:

```
! ERROR:  function _find_circle(geometry, geometry, geometry) does not exist
! LINE 65:             ST_Force_2D(_Find_Circle(n.p1,n.p2,n.p3)) as ct ...
!                                  ^
! HINT:  No function matches the given name and argument types. You might need to add explicit type casts.
```
Because of:
```sql
CREATE OR REPLACE FUNCTION _Find_Circle(
    IN p_pt1 geometry,
    IN p_pt2 geometry,
    IN p_pt3 geometry)
...
```

**Comment by rafatower** [comment]
I must insist in the `search_path` thing. Either you prepend `crankshaft.` to the functions or tweak the search path in your tests (I think there are some tests doing so)

**Comment by AbelVM** [comment]
OK, even with "internal functions"

**Comment by AbelVM** [comment]
Green light!

File: src/pg/sql/09_voronoi.sql [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+   v_dF     NUMERIC;
+   v_dG     NUMERIC;
+BEGIN
+   IF ( p_pt1 IS NULL OR p_pt2 IS NULL OR p_pt3 IS NULL ) THEN
+      RAISE EXCEPTION 'All supplied points must be not null.';
+      RETURN NULL;
+   END IF;
+   IF ( ST_GeometryType(p_pt1) <> 'ST_Point' OR
+        ST_GeometryType(p_pt1) <> 'ST_Point' OR
+        ST_GeometryType(p_pt1) <> 'ST_Point' ) THEN
```
Comment (rafatower) [inline src/pg/sql/09_voronoi.sql]: shouldn't these be checking pt2 an pt3? I'll permit myself to change those

**Comment by AbelVM** [comment]
Cool! Tx Rafa!

**Pull request merged**